include/CL/sycl/common_functions.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * This file is part of hipSYCL, a SYCL implementation based on CUDA/HIP
+ *
+ * Copyright (c) 2019 hipSYCL contributors
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef HIPSYCL_COMMON_FUNCTIONS_HPP
+#define HIPSYCL_COMMON_FUNCTIONS_HPP
+
+#include <type_traits>
+
+#include "vec.hpp"
+#include "math.hpp"
+
+namespace cl {
+namespace sycl {
+
+namespace detail {
+  template<class T>
+  constexpr T pi = T(3.1415926535897932385L);
+}
+
+template<typename T>
+inline auto clamp(T x, T minval, T maxval) {
```

Comment:
please mark all functions intended to be used inside kernels with `HIPSYCL_KERNEL_TARGET` (expands to `__host__ __device__` currently). This is required for the support of "manual mode" and "legacy source-to-source transformation", although the latter is deprecated. See [here](https://github.com/illuhad/hipSYCL/issues/42#issuecomment-483312569) for more details.